argilla/src/argilla/settings/_io/_hub.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2024-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import httpx
+import warnings
+from collections import defaultdict
+from enum import Enum
+from typing import Any, Dict, TYPE_CHECKING, Union, List, Optional
+
+from argilla._exceptions._hub import DatasetsServerException
+from argilla._exceptions._settings import SettingsError
+
+if TYPE_CHECKING:
+    from argilla import Settings
+
+DATASETS_SERVER_BASE_URL = "https://datasets-server.huggingface.co"
+DATASETS_SERVER_HEADERS = {"Accept": "application/json"}
+DATASETS_SERVER_TIMEOUT = 30
+
+
+class FeatureType(Enum):
+    TEXT = "text"
+    CHAT = "chat"
+    IMAGE = "image"
+    LABEL = "label"
+    INT = "int"
+    FLOAT = "float"
+
+
+class AttributeType(Enum):
+    QUESTION = "question"
+    FIELD = "field"
+    METADATA = "metadata"
+
+
+def _get_dataset_features(repo_id: str) -> Dict[str, Dict[str, Any]]:
+    """Get the features of a dataset from the datasets server using the repo_id and config.
+    Extract the features from the response and return them as a dictionary.
+
+    Parameters:
+        repo_id (str): The repository ID of the dataset.
+
+    Returns:
+        Dict[str, Dict[str, Any]]: The features of the dataset.
+    """
+
+    url = f"{DATASETS_SERVER_BASE_URL}/info?dataset={repo_id}"
+
+    try:
+        with httpx.Client(timeout=DATASETS_SERVER_TIMEOUT) as client:
```

Comment:
Just a comment here: 

We can preset some values here to avoid repeat on each call (but we only have one, :_))
```suggestion
        with httpx.Client(base_url= DATASETS_SERVER_BASE_URL, headers=DATASETS_SERVER_HEADERS, timeout=DATASETS_SERVER_TIMEOUT) as client:
```